sdk/core/switcher/entry.S

```diff
@@ -347,11 +398,20 @@ exception_entry_asm:
 	csc                ct0, TrustedStack_offset_mepcc(csp)
 	csrr               t1, mstatus
 	csw                t1, TrustedStack_offset_mstatus(csp)
-#ifdef CONFIG_MSHWM
+#ifdef CHERIOT_HAS_MSHWM
 	csrr               t1, CSR_MSHWM
 	csw                t1, TrustedStack_offset_mshwm(csp)
 	csrr               t1, CSR_MSHWMB
 	csw                t1, TrustedStack_offset_mshwmb(csp)
+#endif
+#ifdef CHERIOT_HAS_ZTOP
+	// Stop zeroing and capture the current zeroing value.  Note: cspecialr is
+	// encoded as cspecialrw and so we can't just use cnull as the source.  We
+	// probably could use the current ct1 value, since it's guaranteed to be
+	// untagged at this point in the code.
```

Comment:
Can we leave the zeroizer running until the `cspecialw` below (making that a load of the new thing, a `cspecialrw`, and a store of the old thing)?  We'd need to keep at least access to `TrustedStack_offset_ztop(csp)` (for *this* `csp`) around, but we have ample registers in which to do so.

Reply:
I pondered that, but I don't like the idea of slowing down the scheduler, since that impacts interrupt latency (switching to a high priority thread on interrupt will be slower if the interrupted thread is doing zeroing).  We can't leave it in a register while we invoke the scheduler and so we'd have to get it back, which might break the code that handles the double fault case.

Reply:
That's a good point about the slowdown, but I'd still kind of like to measure it, at least to know.

But yeah, juggling things around the (untrusted) scheduler is more of a challenge, yes. :(